src/main/java/singly_linked_list/ListService.java

```
@@ -1,33 +1,216 @@
+                cur = null;
+                return top;
+            }
+            prev = cur;
+            cur = cur.next;
+        }
         return null;
     }
 
-    public void insertBefore(Node top, int toInsert, int elementBefore) {
+    @Override
+    public Node insertBefore(Node top, int toAdd, int before) {
+        Node cur = top.next;
+        Node prev = top;
+        Node additional = new Node();
+        additional.info = toAdd;
+        if(top.info == before){
+            top = additional;
+            top.next = prev;
+            return top;
+        }
+        while(cur != null){
+            if(cur.info == before) {
+                prev.next = additional;
+                additional.next = cur;
+                return top;
+            }
+            prev = cur;
+            cur = cur.next;
+        }
+        System.out.println("There is no such number (" + before +
+                ") in the sequence before which you want to put a number " + toAdd);
+        return top;
+    }
+
+
+    @Override
+    public Node insertAfter(Node top, int toAdd, int after) {
+        Node cur = top.next;
+        Node prev = top;
+        Node additional = new Node();
+        additional.info = toAdd;
+        while(cur != null){ // Block, if you need to insert an element somewhere inside the sequence
+            if(prev.info == after) {
+                prev.next = additional;
+                additional.next = cur;
+                return top;
+            }
+            prev = cur;
+            cur = cur.next;
+        }
+        if(prev.info == after){ // Checking if an element needs to be added at the very end (after the last number in the sequence)
+            prev.next = additional;
+            return top;
+        }
+        //If you do not find a number in the sequence, after which you need to add another element, then it displays a message and the original list.
+        System.out.println("There is no such number (" + after +
```

Please throw an exception here too

---

Done.